scripts/tutorial-uploader/lib/api.ts

```diff
@@ -0,0 +1,299 @@
+   * "tutorials" to "tutorials_topics".
+   */
+  async clearTopics(tutorialId: string) {
+    // "tutorials_tutorials_topics" is mapping of tutorial to topics
+    const ids = await this.getIds(
+      "tutorials_tutorials_topics",
+      "tutorials_id",
+      tutorialId,
+    );
+    for (const id of ids) {
+      await this.client
+        .request(deleteItem("tutorials_tutorials_topics", id))
+        .catch((err: any) => console.log(err));
+    }
+  }
+
+  async updateTutorialTopics(tutorialId: string, topicNames: string[]) {
+    await this.clearTopics(tutorialId);
+    for (const name of topicNames) {
+      const id = await this.getId("tutorials_topics", "name", name);
+      if (id === null) throw new Error(`No topic with name '${name}'`);
+      await this.client.request(
+        createItem("tutorials_tutorials_topics", {
+          tutorials_id: tutorialId,
+          tutorials_topics_id: id,
+        }),
+      );
+    }
+  }
+
+  /* Returns the file's ID */
+  async uploadLocalFolder(path: string): Promise<string> {
+    // Zip folder
+    const zippedFilePath = `${tmpdir()}/${randomBytes(8).toString(
+      "hex",
+    )}/tutorial.zip`;
+    await $`mkdir -p ${dirname(zippedFilePath)}`;
+    // We cd into the parent dir to avoid including the full directory
+    // structure in the zip
+    await $`(cd ${dirname(path)} && zip -qr ${zippedFilePath} ${basename(
+      path,
+    )})`;
+
+    // Build form
+    const file = new Blob([await readFile(zippedFilePath)], {
+      type: "application/zip",
+    });
+    const formData = new FormData();
+    formData.append("title", zippedFilePath);
+    formData.append("file", file, zippedFilePath);
+
+    // Upload form
+    const response = await this.client.request(uploadFiles(formData));
+
+    // Clean up
+    await $`rm -r ${dirname(zippedFilePath)}`;
+
+    return response.id;
+  }
+
+  async updateExistingTutorial(
+    tutorialId: string,
+    localData: LocalTutorialData,
+  ) {
+    console.log(`Updating tutorial '${localData.slug}'...`);
+    const newTutorial = {
+      category: await this.getId(
+        "tutorials_categories",
+        "name",
+        localData.category,
+      ),
+      reading_time: localData.reading_time,
+      catalog_featured: localData.catalog_featured,
+      status: localData.status,
+      translations: [
+        {
+          title: localData.title,
+          id: await this.getEnglishTranslationId(tutorialId),
+          temporal_file: await this.uploadLocalFolder(localData.local_path),
+          short_description: localData.short_description,
+        },
+      ],
+    };
+
+    await this.client.request(updateItem("tutorials", tutorialId, newTutorial));
+    await this.updateTutorialTopics(tutorialId, localData.topics);
+  }
+
+  /*
+   * Only sets minimum data required for API to accept the creation request
+   * updateExistingTutorial should be called immediately after
```

Comment:
Idea prompted by this usage note: make most the functions private by putting `#` in front of the name so that it's clear what you expect other files to use or not. That reduces the risk of someone encountering this "gotcha" that they need to also use updateExistingTutorial